src/components/calcite-input/calcite-input.tsx

```diff
@@ -586,6 +594,7 @@ export class CalciteInput {
         disabled={this.disabled}
         onClick={this.clearInputValue}
         tabIndex={this.disabled ? -1 : 0}
+        type="button"
```

Comment:
Do you think we'll remember to remove this once `calcite-input` is shadowified™?

Reply:
Its probably a best practice for most of internal buttons to set the type anyway so that they don't submit a form. Its ok if it stays as button type. It won't cause any harm at all.